app/services/jokes.js

```diff
@@ -11,12 +11,13 @@ const {
 exports.getRandomJoke = () => {
   info('Calling jokesService.getRandomJoke');
   return axios
-    .get(baseUrl)
+    .get(baseUrl, { params: { format: 'json' } })
     .then(response => {
       if (response.data && response.data.joke) return response.data.joke;
       throw externalServiceError('Geek Jokes responds with an invalid body');
     })
     .catch(error => {
+      if (error.internalCode && error.internalCode === EXTERNAL_SERVICE_ERROR) throw error;
```

Comment:
If you don't want to catch the error thrown in the previous then, you could move the catch before the .then